ceno_zkvm/src/instructions/riscv/dummy/dummy_circuit.rs

```diff
@@ -0,0 +1,238 @@
+use std::marker::PhantomData;
+
+use ceno_emul::{InsnCategory, InsnCodes, InsnFormat, InsnKind, StepRecord};
+use ff_ext::ExtensionField;
+
+use super::super::{
+    RIVInstruction,
+    constants::UInt,
+    insn_base::{ReadMEM, ReadRS1, ReadRS2, StateInOut, WriteMEM, WriteRD},
+};
+use crate::{
+    circuit_builder::CircuitBuilder,
+    error::ZKVMError,
+    expression::{ToExpr, WitIn},
+    instructions::Instruction,
+    set_val,
+    tables::InsnRecord,
+    uint::Value,
+    utils::i64_to_base,
+    witness::LkMultiplicity,
+};
+use core::mem::MaybeUninit;
+
+/// DummyInstruction can handle any instruction and produce its side-effects.
+pub struct DummyInstruction<E, I>(PhantomData<(E, I)>);
+
+impl<E: ExtensionField, I: RIVInstruction> Instruction<E> for DummyInstruction<E, I> {
+    type InstructionConfig = DummyConfig<E>;
+
+    fn name() -> String {
+        format!("{:?}_DUMMY", I::INST_KIND)
+    }
+
+    fn construct_circuit(
+        circuit_builder: &mut CircuitBuilder<E>,
+    ) -> Result<Self::InstructionConfig, ZKVMError> {
+        DummyConfig::construct_circuit(circuit_builder, I::INST_KIND.codes())
+    }
+
+    fn assign_instance(
+        config: &Self::InstructionConfig,
+        instance: &mut [MaybeUninit<<E as ExtensionField>::BaseField>],
+        lk_multiplicity: &mut LkMultiplicity,
+        step: &StepRecord,
+    ) -> Result<(), ZKVMError> {
+        config.assign_instance(instance, lk_multiplicity, step)
+    }
+}
+
+#[derive(Debug)]
+pub struct DummyConfig<E: ExtensionField> {
+    vm_state: StateInOut<E>,
+
+    rs1: Option<(ReadRS1<E>, UInt<E>)>,
+    rs2: Option<(ReadRS2<E>, UInt<E>)>,
+    rd: Option<(WriteRD<E>, UInt<E>)>,
+
+    mem_addr_val: Option<[WitIn; 3]>,
+    mem_read: Option<ReadMEM<E>>,
+    mem_write: Option<WriteMEM>,
+
+    imm: WitIn,
+}
+
+impl<E: ExtensionField> DummyConfig<E> {
+    fn construct_circuit(
+        circuit_builder: &mut CircuitBuilder<E>,
+        codes: InsnCodes,
+    ) -> Result<Self, ZKVMError> {
+        let (with_rs1, with_rs2, with_rd) = match (codes.format, codes.kind) {
+            (_, InsnKind::EANY) => (true, true, false),
+            (InsnFormat::R, _) => (true, true, true),
+            (InsnFormat::I, _) => (true, false, true),
+            (InsnFormat::S, _) => (true, true, false),
+            (InsnFormat::B, _) => (true, true, false),
+            (InsnFormat::U, _) => (false, false, true),
+            (InsnFormat::J, _) => (false, false, true),
```

Comment:
Maybe we can get rid of them as opcodes that are not supported (i.e. `DIV / REM / REMU`) right now belong to the `R` category.

Reply:
Yes we have implemented almost everything in the mean time 😄 . Maybe there will be other use-cases: test, benchmarks, demos ?